### ISSUE TITLE
How act on scope of Dotfiles?

### ISSUE THREAD

**Issue by florianb** [atom/autocomplete-plus#789]
Hi, i am currently working on an autocomplete-provider for `.editorconfig`-files and as it is everywhere described, i created a provider for the scope-selector `.source.editorconfig` (a grammar we also provide).

Unfortunately the `getSuggestions()`-function is not called for any `.editorconfig` file, whereas it is working fine for `abcd.editorconfig`-files.

I would be very thankful if anybody could tell me if autocomplete-plus should be able to work on dotfiles and if so, how the source-selector should look like?

I promise i'll put my knowledge into the provider-api's wiki article.. 🍻

**Comment by isiahmeadows** [comment]
Actually, there's a file blacklist that defaults to `.*`, but could easily be modified. One thing it should allow is negation, since minimatch doesn't support that by default IIRC.

**Comment by florianb** [comment]
So i guess changing

    if (minimatch(fileName, blacklistGlob)) {

at [this](https://github.com/atom/autocomplete-plus/blob/b5ed0c76b21edd27a10ecc9dc23f69397b9883fe/lib/autocomplete-manager.js#L582) position to

    if (minimatch(fileName, blacklistGlob, {dot: true})) {

could fix this according, to the [minimatch-readme](https://github.com/isaacs/minimatch).

**Comment by isiahmeadows** [comment]
Not exactly. That affects things like `foo/*` and `foo/**` (whether those
match dot files). The default glob of `.*` will trigger dot file matching
regardless, because it has an explicit dot.
